### PR TITLE
Update reader to import all dataset vars by default

### PR DESCRIPTION
**Pull request by biosafetylvl5**
# Update CLAVR-x Reader to Pass Through All Attributes and Some Other Stuff

## Semi-relevant XKCD

i promise!!! this time

* [x] Required ***existing tests*** pass (ie full_test.sh, others as appropriate)
* [x] Required ***unit tests*** added and pass for new/modified functionality
* [x] Required ***integration tests*** added and pass for new/modified functionality
* [x] Required ***documentation*** added for new/modified functionality
* [x] Required ***release notes*** added for new/modified functionality
* [x] NO REQUIRED ***updates to other repos***

## Related Issues
fixes NRLMMD-GEOIPS/geoips-clavrx#49

## Summary

- **Attribute Passing**: Modified the reader to accept and pass along all attributes from the original data file in the exported data.
- **Coordinate Addition**: Added coordinate values (`coords`) to each of the sub-data arrays to support downstream scripts that rely on this structure.
- **Refactoring and Documentation**:
  - Refactored the reading code for improved readability and maintainability.
  - Added/updated docstrings
  - Added some logging for debugging.
- **Scaling Attributes Removal**: Stripped scaling attributes (`scale_factor`, `add_offset`, etc.) from the output data arrays since the data is scaled within the code and these become meaningless
- **Documentation Build Fixes**: Updated docstrings and release note files to ensure the documentation builds successfully without errors.
- **Docker**: Adds .devcontainer and Dockerfile for future development

## Motivation

This is required for a downstream plugin that needs to access an attribute that was previously not passed.

## Testing

- [x] Unit tests pass
- [x] base_test pass
- [x] full_test pass

### CODE REVIEW

**Comment by biosafetylvl5** [comment]
@AJThorpe I think so! It's passing all the integration tests for me. Let me know if you see anything on your tests

File: .devcontainer/devcontainer.json
Comment (mindyls) [inline .devcontainer/devcontainer.json:3]: Do not merge this until pointing to main

File: docs/source/releases/latest/read-all-vars-by-default.yaml
Comment (mindyls) [inline docs/source/releases/latest/read-all-vars-by-default.yaml:24]: isort is the only change that is called out in this release note.  Seems like there are A LOT of changes in this PR, including dev container, refactoring code, black updates ?  I would like a little more clarity on what's changed.

Comment (biosafetylvl5) [reply docs/source/releases/latest/read-all-vars-by-default.yaml:24]: Oh woops! Thanks mindy. I forgot to add the `enhancement` portion. I'll add that and then ping you again.

File: .devcontainer/devcontainer.json
Comment (biosafetylvl5) [reply .devcontainer/devcontainer.json:3]: Merging this into the `multi-scan-time` PR first could be helpful because it fixes some docstrings/release notes in that PR that were misformatted and thus caused the CI to fail

Comment (biosafetylvl5) [reply .devcontainer/devcontainer.json:3]: But waiting until it points at main is also fine; either works

File: docs/source/releases/latest/read-all-vars-by-default.yaml
Comment (biosafetylvl5) [reply docs/source/releases/latest/read-all-vars-by-default.yaml:24]: Fixed.

File: .devcontainer/devcontainer.json
Comment (mindyls) [reply .devcontainer/devcontainer.json:3]: That would make it more difficult to review the other PR, I'd rather merge it with a few actions failing, knowing that all actions will pass on this PR before merging this one.

File: .devcontainer/devcontainer.json
Comment (mindyls) [inline .devcontainer/devcontainer.json:18]: Should this be generalized?

File: .devcontainer/devcontainer.json
Comment (biosafetylvl5) [reply .devcontainer/devcontainer.json:3]: Works for me! Whatever's easiest to review :)

File: Dockerfile
Comment (biosafetylvl5) [reply Dockerfile:7]: They're from the settings in the `geoips` Dockerfile that builds the base image

Comment (biosafetylvl5) [reply Dockerfile:7]: But beyond that, yeah; they're arbitrary and just need to be the same

File: .devcontainer/devcontainer.json
Comment (biosafetylvl5) [reply .devcontainer/devcontainer.json]: Yeah, you're right. That should be generalized

Comment (biosafetylvl5) [reply .devcontainer/devcontainer.json]: Fixed!

File: Dockerfile
Comment (jsolbrig) [reply Dockerfile:7]: @mindyls UID and GID inside containers are typically fairly arbitrary and, at runtime, a different set can be used. This looks good to me.

File: geoips_clavrx/plugins/modules/readers/clavrx_hdf4.py
Comment (jsolbrig) [inline geoips_clavrx/plugins/modules/readers/clavrx_hdf4.py:63]: Why is this only required for ABI?

File: geoips_clavrx/plugins/modules/readers/clavrx_hdf4.py
Comment (jsolbrig) [inline geoips_clavrx/plugins/modules/readers/clavrx_hdf4.py:125]: Just a note, but this set of variables can change based on how CLAVR-x is configured. I'm not sure that this is the full set of variables, just what happens to be available in the files we're currently using. Even if it is the full current set, new variables sometimes get added to CLAVR-x.

I'm not sure what I'm suggesting. Just pointing out that this may or may not be an exhaustive list.

**Review by jsolbrig — APPROVED** [review]

File: geoips_clavrx/plugins/modules/readers/clavrx_hdf4.py
Comment (biosafetylvl5) [reply geoips_clavrx/plugins/modules/readers/clavrx_hdf4.py:63]: I don't know; that's from Songs previous work. I figured it was some kind of historical/anomalous thing

**Pull request merged**
